packages/nodes-base/nodes/Redis/Redis.node.ts

```diff
@@ -630,6 +769,48 @@ export class Redis implements INodeType {
 								const clientPublish = util.promisify(client.publish).bind(client);
 								await clientPublish(channel, messageData);
 								returnItems.push(items[itemIndex]);
+							} else if (operation === 'push'){
+								const redisList = this.getNodeParameter('list', itemIndex) as string;
+								const messageData = this.getNodeParameter('messageData', itemIndex) as string;
+								const tail = this.getNodeParameter('tail', itemIndex, false) as boolean;
+
+								const action = tail ? client.rpushx : client.lpushx;
+								const clientPush = util.promisify(action).bind(client);
+
+								await clientPush(redisList, messageData);
+								throw Error;
```

Comment:
This will never return successfully as we are throwing an error.

Reply:
Also, looks like the push operation it's not a working property. When I push something via the node (I removed the throw error line locally) and try to pop it, it always returns null. However, when I push data via de CLI and pop it with the node it works fine.

Reply:
My bad, not sure how that got in after testing! I seem to be getting back the input data with this example, similar to the existing `publish` action:

```json
{
  "nodes": [
    {
      "parameters": {
        "functionCode": "items = [];\n\nfor (let i=0; i<10; i++) {\n    items.push(\n      {key: `val${i}`}\n    )\n}\n\nreturn items;"
      },
      "name": "Function",
      "type": "n8n-nodes-base.function",
      "typeVersion": 1,
      "position": [
        240,
        80
      ]
    },
    {
      "parameters": {
        "operation": "push",
        "list": "key",
        "messageData": "={{$json.key}}"
      },
      "name": "LPUSH",
      "type": "n8n-nodes-base.redis",
      "typeVersion": 1,
      "position": [
        460,
        80
      ],
      "credentials": {
        "redis": {
          "id": "11",
          "name": "Redis account"
        }
      }
    }
  ],
  "connections": {
    "Function": {
      "main": [
        [
          {
            "node": "LPUSH",
            "type": "main",
            "index": 0
          }
        ]
      ]
    }
  }
}
```

Reply:
Also the null keys were likely because the list didn’t exist and I was using PUSHX operations. It now uses plain PUSH operations which will create an empty list if one doesn’t exist. Which is more likely what is expected.

Reply:
> My bad, not sure how that got in after testing!

@pemontto , no I think it was my bad, miss it when cleaning up after testing